src/Microsoft.AspNet.Server.Kestrel/Http/Frame.StateMachine.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http
+{
+    public partial class Frame
```

Comment:
How hard would it be to move this logic into its own class? I really like having the state machine logic in its own file, but its own class would be even better.